crates/volta-core/src/manifest/mod.rs

```diff
@@ -96,13 +100,18 @@ impl Manifest {
             }
 
             // serialize the updated contents back to package.json
-            let file = File::create(&package_file)
+            let mut file = File::create(&package_file)
                 .with_context(|_| ErrorDetails::PackageWriteError { file: get_file() })?;
             let formatter =
                 serde_json::ser::PrettyFormatter::with_indent(indent.indent().as_bytes());
-            let mut ser = serde_json::Serializer::with_formatter(file, formatter);
+            let mut ser = serde_json::Serializer::with_formatter(&file, formatter);
             map.serialize(&mut ser)
                 .with_context(|_| ErrorDetails::PackageWriteError { file: get_file() })?;
+            // append the empty line if the original package.json has one
+            if is_end_with_newline {
+                file.write_all("\n".as_bytes())
```

Comment:
Instead of using `write_all`, this is a good spot to use the `write!` macro:

```
if is_end_with_newline {
    write!(file, "\n").with_context(...)?;
}
```

Reply:
cool updated the PR with `write!`, thanks for the suggestion 👍